db/migrate/20230208185106_add_tsv_to_posts.rb

```diff
@@ -0,0 +1,42 @@
+class AddTsvToPosts < ActiveRecord::Migration[6.0]
+  def up
+    add_column :posts, :tsv, :tsvector
+    add_index :posts, :tsv, using: :gin
+
+    execute <<-SQL
+      CREATE FUNCTION update_tsv_on_post() RETURNS trigger LANGUAGE plpgsql AS $$
+        declare
+          thread_title character varying;
+        begin
+          SELECT title INTO thread_title FROM discussion_threads WHERE id = new.thread_id;
+
+          new.tsv := to_tsvector('pg_catalog.english', coalesce(new.body, '')) ||
+                     to_tsvector('pg_catalog.english', coalesce(thread_title, ''));
+          return new;
+        end
+      $$
+    SQL
+
+    execute <<-SQL
+      CREATE TRIGGER update_tsv_on_post BEFORE INSERT OR UPDATE ON posts FOR EACH ROW EXECUTE FUNCTION update_tsv_on_post()
```

Comment:
We assume that the thread title never changes, and the first post is always created after the thread so we don't bother adding a trigger+function pair when threads are created/updated.